src/Microsoft.OpenApi.Readers/Document.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Readers
+{
+    /// <summary>
+    /// Loads an OpenApiDocument instance through Load/LoadAsync/Parse pattern
+    /// </summary>
+    public static class Document
+    {
+        /// <summary>
+        /// Loads an OpenApiDocument from a file stream
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="diagnostic"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static OpenApiDocument Load(Stream stream, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null)
```

Comment:
These should be extension methods so that they appear on the OpenApiDocument class.

Reply:
we don't have an instance of OpenAPIDocument yet, and as far as I know static extension methods are not a thing. Maybe the design needs to be reviewed here?

Reply:
> These should be extension methods so that they appear on the OpenApiDocument class.

So the format should ideally look like this: 
```c#
var doc = new OpenApiDocument().Load(stream)
```

Reply:
in which case we wouldn't be returning the document from the method? Works for me but I'm not sure the internals of OpenAPIDocument are accessible in an extension method outside of the assembly.

Reply:
This would create an odd pattern of
``` csharp
var document = new OpenApiDocument();

document.Load();
```

Where document would now be null.


Perhaps marking OpenApiDocument as partial could do the trick as well (adding the static load methods in the reader lib)

Reply:
thanks for jumping in the discussion. The partial solution would only work if the classes were part of the same assembly and namespace. But these methods need to be in the readers package since they have dependency on readers. Whereas the OpenAPIDocument class is in the main package.

Reply:
Forgot about the assembly part.

Reply:
> in which case we wouldn't be returning the document from the method? Works for me but I'm not sure the internals of OpenAPIDocument are accessible in an extension method outside of the assembly.

We can return a document in order to avoid null references when we return a void as @VisualBean has indicated [here](https://github.com/microsoft/OpenAPI.NET/pull/1446#discussion_r1377830123).

Reply:
This pattern is strange though. Why define those methods as extension methods if we're returning a new instance? How is the consumer supposed to know which instance to hold on to?

Reply:
Perhaps simply making static methods on the readers will provide an adequate solution.

The main issue (compared to XmlDocument) is that reading is separate from the models.

The main problem Darrel seems to reference is that the readers aren't really 'readers', they simply parse the entirety of a string/stream/whatever and make a document (or fragment), so newing them seems wrong, as they don't keep state (other than options).